cmd/snap-update-ns/utils.go

```diff
@@ -98,22 +98,36 @@ func (sec *Secure) OpenPath(path string) (int, error) {
 	// We use the following flags to open:
 	//  O_PATH: we don't intend to use the fd for IO
 	//  O_NOFOLLOW: don't follow symlinks
-	//  O_DIRECTORY: we expect to find directories
+	//  O_DIRECTORY: we expect to find directories (except for the leaf)
 	//  O_CLOEXEC: don't leak file descriptors over exec() boundaries
-	const openFlags = sys.O_PATH | syscall.O_NOFOLLOW | syscall.O_DIRECTORY | syscall.O_CLOEXEC
+	openFlags := sys.O_PATH | syscall.O_NOFOLLOW | syscall.O_DIRECTORY | syscall.O_CLOEXEC
 	var fd int
 	fd, err = sysOpen("/", openFlags, 0)
 	if err != nil {
 		return -1, err
 	}
-	for _, segment := range segments {
+	for i, segment := range segments {
 		// Ensure the parent file descriptor is closed
 		defer sysClose(fd)
+		if i == len(segments)-1 {
```

Comment:
We are dropping O_DIRECTORY unconditionally on the last element when we don't strictly have to. This is not a blocker but wanted to call it out as a change.